lib/apiservers/engine/backends/cache/image_cache.go

```diff
@@ -115,3 +119,51 @@ func (c *ImageCache) GetImages() []*metadata.ImageConfig {
 
 	return result
 }
+
+func (c *ImageCache) GetImageByDigest(digest digest.Digest) *metadata.ImageConfig {
+	c.m.RLock()
+	defer c.m.RUnlock()
+
+	var imageID string
+
+	for _, image := range c.cache {
```

Comment:
This loop is currently O(n) for the number of images we have. If lookup by digest is needed in anything other than very rare cases, consider having an additional index keyed by digest.